src/iso.c

```diff
@@ -0,0 +1,489 @@
+/*
+Copyright (C) 2016 The University of Notre Dame
+This software is distributed under the GNU General Public License.
+See the file LICENSE for details.
+*/
+
+/*
+ * This file has external facing iso_f and iso_d commands, but also
+ * contains iso_media_ commands for open, read, and write. The idea is
+ * that iso_f and iso_d are in regards to files and directories, while
+ * iso_media_ treats the entire iso storage device as a single, long file
+ * without an EOF which we can open, close, and seek around. rather than
+ * doing absurd byte counting in each of the iso_f and iso_d commands.
+ */
+
+#include "iso.h"
+#include "memory.h"
+#include "string.h"
+#include "kerneltypes.h"
+#include "ata.h"
+#include "console.h"
+#include "keyboard.h"
+
+#define ISO_BLOCKSIZE 2048
+#define PVD_OFFSET 16 * ISO_BLOCKSIZE
+#define ROOT_DR_OFFSET (PVD_OFFSET) + 156
+
+#define SEEK_CUR 0
+#define SEEK_SET 1
+
+// Use these to prevent multiple fetches of the same
+// ATAPI block for consecutive, single character reads
+char global_atapi_block[ATAPI_BLOCKSIZE];
+int global_atapi_unit = -1;
+int global_atapi_extent = -1;
+
+struct iso_point {
+    int ata_unit;
+    int cur_extent;
+    int cur_offset;
+};
+
+int get_directory_record(struct iso_point *iso_p, struct directory_record *dr);
+int hex_to_int(char *src, int len);
```

Comment:
I feel like this function should live somewhere else, since other people may want to use it... Maybe we should start discussing adding a std lib, and put a todo to move it in there later?